# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.h
# Commit: 33dcf5c2e643
# Full Hash: 33dcf5c2e643b94e90a0a197c99112a080501a12
# Author: Alexandra Borovova <aborovova@mozilla.com>
# Date: 2025-08-20 09:57:21
# Regressor Bug: 1980211
# File Overlap Count: 2
# Description:
#   Bug 1980211 - Attach locale override to Realm instead of setting it in the process. r=dom-core,smaug,tschuster
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D260169
# ==============================================================================

diff -r 2cb7ca176216 -r 33dcf5c2e643 docshell/base/BrowsingContext.h
--- a/docshell/base/BrowsingContext.h	Tue Aug 19 16:57:59 2025 +0000
+++ b/docshell/base/BrowsingContext.h	Tue Aug 19 17:01:50 2025 +0000
@@ -240,7 +240,7 @@
   FIELD(MediumOverride, nsString)                                             \
   /* DevTools override for prefers-color-scheme */                            \
   FIELD(PrefersColorSchemeOverride, dom::PrefersColorSchemeOverride)          \
-  FIELD(LanguageOverride, nsString)                                           \
+  FIELD(LanguageOverride, nsCString)                                          \
   /* DevTools override for forced-colors */                                   \
   FIELD(ForcedColorsOverride, dom::ForcedColorsOverride)                      \
   /* prefers-color-scheme override based on the color-scheme style of our     \
@@ -968,7 +968,7 @@
     aOverride = GetMediumOverride();
   }
 
-  void GetLanguageOverride(nsAString& aLanguageOverride) const {
+  void GetLanguageOverride(nsACString& aLanguageOverride) const {
     aLanguageOverride = GetLanguageOverride();
   }
 
@@ -1135,7 +1135,7 @@
     return IsTop();
   }
 
-  bool CanSet(FieldIndex<IDX_LanguageOverride>, const nsString&,
+  bool CanSet(FieldIndex<IDX_LanguageOverride>, const nsCString&,
               ContentParent*) {
     return IsTop();
   }
@@ -1176,7 +1176,7 @@
   void WalkPresContexts(Callback&&);
   void PresContextAffectingFieldChanged();
 
-  void DidSet(FieldIndex<IDX_LanguageOverride>, nsString&& aOldValue);
+  void DidSet(FieldIndex<IDX_LanguageOverride>, nsCString&& aOldValue);
 
   void DidSet(FieldIndex<IDX_MediumOverride>, nsString&& aOldValue);
 